pkg/ruler/remotequerier.go

```diff
@@ -65,6 +68,8 @@ func (c *QueryFrontendConfig) RegisterFlags(f *flag.FlagSet) {
 		"GRPC listen address of the query-frontend(s). Must be a DNS address (prefixed with dns:///) "+
 			"to enable client side load balancing.")
 
+	f.DurationVar(&c.Timeout, "ruler.query-frontend.timeout", 2*time.Minute, "The timeout for a rule query being evaluated by the query-frontend.")
```

Comment:
Do we have any metrics about how long these queries are taking to run now, and how many are exceeding the 2min limit? 
I'm a bit concerned that we might currently be running lots of slow queries without being aware of it, and once we upgrade to this version of Mimir we'll suddenly start to kill all these slow queries.